### DISCUSSION TITLE
lesson 6, RPC url error.

### DISCUSSION THREAD

**Discussion by olahamid** [Cyfrin/foundry-full-course-cu#525] [Q&A] [answered]
i keep getting an error when trying to deploy the simplestorage contraact using foundry just as it's done in the course using foundry, i keep getting the error below i've tried troubleshooting shooting using AI and i've tried turned off the window defendeer firewall(thought it was a firewall restriction error). it didn't work still got the error below. 
can anyone give any tip that can help?
COMMAND
```gitpod /workspace/new_folder2/foundry-f233/foundry-simple-storage-f233 (master) $ forge create SimpleStorage --rpc-url http://127.0.0.1:7545
```
ERROR
```
[⠑] Compiling...
No files changed, compilation skipped
Error: 
error sending request for url (http://127.0.0.1:7545/): error trying to connect: tcp connect error: Connection refused (os error 111)

Context:
- Error #0: error trying to connect: tcp connect error: Connection refused (os error 111)
- Error #1: tcp connect error: Connection refused (os error 111)
- Error #2: Connection refused (os error 111)
```

**Comment by 0xusmanf** [comment, marked as answer]
First run anvil in a separate terminal by running the `anvil` command and then try again. And I think the rpc url for anvil is `http://127.0.0.1:8545`.

**Reply by olahamid** [reply]
I'm having a slight issue setting up anvil ON METAMASK, 31337 doesnt work as the network ID 31337
![Screenshot (94)](https://github.com/Cyfrin/foundry-full-course-f23/assets/108703357/6144c0d9-2127-47f0-a9fd-775f9c011bc2)

**Reply by 0xusmanf** [reply]
Again, run anvil before adding it to metamask.

**Reply by olahamid** [reply]
i ran anvil on my terminal already, worked fine.

**Reply by 0xDevGod** [reply]
I am getting this error
```solidity
script: unrecognized option '--rpc-url'
```

The solution to this as well is deleting your existing terminal and follow along Patrick exactly.

**Reply by SamuelBruno-lab** [reply]
Hi I already ran Anvil and tried to add network again but I get the same error again and again.

(My WSL is on ubuntu and my metamask is on windows)

**Reply by Top88Star** [reply]
[Sengbepieh](https://github.com/Sengbepieh) Reload VSCode, your browser with the MetaMask, Make sure that your terminal is set in the specific folder of your project and then run `anvil`.

**Comment by olahamid** [comment]
works with anvil, thanks alot again usman. 👌

**Reply by 0xusmanf** [reply]
You are welcome :)

**Comment by sahilbansal123-lang** [comment]
I'm also getting this error while trrying to connect to ganache

Command 
```
forge create SimpleStorage --rpc-url HTTP://127.0.0.1:7545 --interactive

```
AND ERROR im getting on terminal is: 

```

error sending request for url (http://127.0.0.1:7545/): error trying to connect: tcp connect error: Connection refused (os error 111)

Context:
- Error #0: error trying to connect: tcp connect error: Connection refused (os error 111)
- Error #1: tcp connect error: Connection refused (os error 111)
- Error #2: Connection refused (os error 111)

```

**Reply by theirrationalone** [reply]
That's PORT: 8545 not 7545, Please Try Again with `http://127.0.0.1:8545`

**Reply by sahilbansal123-lang** [reply]
Still getting this error bro

**Reply by theirrationalone** [reply]
Try with these RPCs...

`http://0.0.0.0:8545`
'http://0.0.0.0:7545`

**Comment by MarcoW3** [comment]
Guys, I've got a similar problem (same error message), but I cannot solve it with any of the RPCs suggested. 


<details><summary>Details</summary>
<p>

COMMAND

```
foundry-simple-storage-f23$forge create SimpleStorage --rpc-url http://127.0.0.1:8545 --interactive
```

ERROR

```
[⠑] Compiling...
No files changed, compilation skipped
Error: 
error sending request for url (http://127.0.0.1:8545/): error trying to connect: tcp connect error: Connection refused (os error 111)

Context:
- Error #0: error trying to connect: tcp connect error: Connection refused (os error 111)
- Error #1: tcp connect error: Connection refused (os error 111)
- Error #2: Connection refused (os error 111)
```

</p>
</details> 

<details><summary>Details</summary>
<p>

COMMAND

```
foundry-simple-storage-f23$ forge create SimpleStorage --rpc-url http://0.0.0.0:8545 --interactive
```

ERROR

```
[⠢] Compiling...
No files changed, compilation skipped
Error: 
error sending request for url (http://0.0.0.0:8545/): error trying to connect: tcp connect error: Connection refused (os error 111)

Context:
- Error #0: error trying to connect: tcp connect error: Connection refused (os error 111)
- Error #1: tcp connect error: Connection refused (os error 111)
- Error #2: Connection refused (os error 111)
```

</p>
</details> 

<details><summary>Details</summary>
<p>

COMMAND

```
foundry-simple-storage-f23$ forge create SimpleStorage --rpc-url http://0.0.0.0:7545 --interactive
```

ERROR

```
[⠑] Compiling...
No files changed, compilation skipped
Error: 
error sending request for url (http://0.0.0.0:7545/): error trying to connect: tcp connect error: Connection refused (os error 111)

Context:
- Error #0: error trying to connect: tcp connect error: Connection refused (os error 111)
- Error #1: tcp connect error: Connection refused (os error 111)
- Error #2: Connection refused (os error 111)
```

</p>
</details> 

Thanks a lot for helping!

**Reply by 0xusmanf** [reply]
First, run anvil in a separate terminal by running the `anvil` command and then try again.

**Reply by MarcoW3** [reply]
Thanks a lot ;)

**Comment by Totiruzi** [comment]
Hello I am trying to connect with Ganache and I am getting the same error
```solidity
forge create SimpleStorage --rpc-url HTTP://127.0.0.0:7545 --interactive
[⠊] Compiling...
No files changed, compilation skipped
Error: 
error sending request for url (http://127.0.0.0:7545/): error trying to connect: tcp connect error: Connection refused (os error 111)

Context:
- Error #0: error trying to connect: tcp connect error: Connection refused (os error 111)
- Error #1: tcp connect error: Connection refused (os error 111)
- Error #2: Connection refused (os error 111)
```

**Reply by Totiruzi** [reply]
I got it working with anvil but not with Ganache!

**Comment by piyushimself** [comment]
I am running avnil in my VPS and want to connect it with Metamask in my desktop. What I need to put in `New RPC URL`

**Reply by EngrPips** [reply]
http://127.0.0.1:8545

**Reply by piyushimself** [reply]
I ran `anvil --host 0.0.0.0 --port 8545` instead of "anvil". Then added `http://VPS-IP:8545` in RPC URL and `31337` as chain ID. It worked.

**Reply by EngrPips** [reply]
Interesting. Keep crushing it.
